src/vm/checker/typecheck/mod.rs

```diff
@@ -268,8 +274,8 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
                     if let Some(Some(ref expected)) = self.function_return_tracker {
                         // check if the computed return type matches the return type
                         //   of any early exits from the call graph (e.g., (expects ...) calls)
-                        TypeSignature::most_admissive(expected.clone(), return_type)
-                            .map_err(|_| CheckError::new(CheckErrors::ReturnTypesMustMatch))?
+                        TypeSignature::most_admissive(expected.clone(), return_type.clone())
+                            .map_err(|_| CheckError::new(CheckErrors::ReturnTypesMustMatch(expected.clone(), return_type)))?
```

Comment:
Same as above -- `most_admissive` returns its args in the error to avoid clones.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1041/commits/dfacb432b383eb88f9c6df483e54466f3d29bf3e